chainntnfs/btcdnotify/btcd.go

```diff
@@ -348,30 +348,41 @@ out:
 			update := item.(*chainUpdate)
 			if update.connect {
 				if update.blockHeight != b.bestBlock.Height+1 {
-					chainntnfs.Log.Warnf("Received blocks out of order: "+
-						"current height=%d, new height=%d",
-						currentHeight, update.blockHeight)
-					continue
-				}
+					// Handle the case where the notifier
+					// missed some blocks from its chain
+					// backend
+					chainntnfs.Log.Infof("Missed blocks, " +
+						"attempting to catch up")
+					newBestBlock, missedBlocks, err :=
+						chainntnfs.HandleMissedBlocks(
+							b.chainConn,
+							b.txConfNotifier,
+							b.bestBlock,
+							update.blockHeight,
+							true,
+						)
+					// Set the bestBlock here in case
+					// a catch up partially completed.
+					b.bestBlock = newBestBlock
```

Comment:
should we do this only in the error case (for readability), since in the success case it should be set later in `handleBlockConnected`?

Reply:
fixed